python/shark_turbine/aot/support/ir_utils.py

```diff
@@ -221,3 +228,21 @@ def emit_return(self, *ir_values: Value):
             ftype = FunctionType.get(ftype.inputs, value_types)
             self.func_op.attributes["function_type"] = TypeAttr.get(ftype)
             assert self.func_op.verify(), "Created function is invalid"
+
+
+###############################################################################
+# Helpers
+###############################################################################
+
+
+def build_index_attribute(value: int) -> IntegerAttr:
+    return IntegerAttr.get(IndexType.get(), value)
+
+
+def build_index_value(value: int) -> Value:
+    attr = build_index_attribute(value)
+    return arith_d.ConstantOp(attr.type, attr).result
+
+
+def build_tensor_dim_value(t: Value, dim: int) -> Value:
+    return tensor_d.DimOp(t, build_index_value(dim)).result
```

Comment:
You should be able to pass dim here directly due to https://github.com/llvm/llvm-project/blob/3017545e6337d37e947a18cb82cf50ec2e7eb3ba/mlir/python/mlir/ir.py#L34

Reply:
I think you meant a few lines up in build_index_value. Indeed, that works... thanks.